### PR TITLE
Add feed of following users

### PR DESCRIPTION
**Pull request by rofiuddin**

### CODE REVIEW

File: app/controllers/static_pages_controller.rb [hunk 1]
```diff
@@ -1,4 +1,7 @@
 class StaticPagesController < ApplicationController
   def home
+    if logged_in?
+      @feed_items = current_user.feed.paginate(page: params[:page])
```
Comment (chuanhtuan) [inline app/controllers/static_pages_controller.rb]: ()

File: app/models/user.rb [hunk 2]
```diff
@@ -10,4 +10,14 @@ class User < ActiveRecord::Base
                     uniqueness: { case_sensitive: false }
   has_secure_password
   validates :password, presence: true, length: { minimum: 8 }
+
+  scope :total_progress, ->(user_id) do
+    lesson = Lesson.where("lessons.user_id= ?", user_id)
+    lesson.sum(:progress)
```
Comment (chuanhtuan) [inline app/models/user.rb]: what is the progress in lesson table

Comment (rofiuddin) [inline app/models/user.rb]: it is result of a lesson

File: app/models/user.rb [hunk 3]
```diff
@@ -10,4 +10,14 @@ class User < ActiveRecord::Base
                     uniqueness: { case_sensitive: false }
   has_secure_password
   validates :password, presence: true, length: { minimum: 8 }
+
+  scope :total_progress, ->(user_id) do
```
Comment (chuanhtuan) [inline app/models/user.rb]: I really do not understand what do you want in this scope
:(

File: app/models/user.rb [hunk 4]
```diff
@@ -10,4 +10,14 @@ class User < ActiveRecord::Base
...
+    lesson = Lesson.where("lessons.user_id= ?", user_id)
```
Comment (chuanhtuan) [inline app/models/user.rb]: Lesson.where("lessons.user_id= ?", user_id).sum(:progress)
It's enough

File: app/views/shared/_stats.html.erb [hunk 5]
```diff
@@ -4,12 +4,18 @@
...
   <a>
     <strong id="followers" class="stat">
       <%= @user.followers.count %>
     </strong>
-    followers
+    <%= t('stat.followers')%>
+  </a>
+  <a>
+    <strong id="progress" class="stat">
+      <%= User.total_progress(@user) %>
```
Comment (chuanhtuan) [inline app/views/shared/_stats.html.erb]: current_user.lessons.sum :progress

File: app/models/user.rb [hunk 6]
```diff
@@ -10,4 +10,10 @@ class User < ActiveRecord::Base
                     uniqueness: { case_sensitive: false }
   has_secure_password
   validates :password, presence: true, length: { minimum: 8 }
+
+  def feed
+    following_ids_subselect = self.following_ids
```
Comment (chuanhtuan) [inline app/models/user.rb]: please see again Chapter 12 in tutorial

File: app/models/user.rb [hunk 7]
```diff
@@ -10,4 +10,11 @@ class User < ActiveRecord::Base
                     uniqueness: { case_sensitive: false }
   has_secure_password
   validates :password, presence: true, length: { minimum: 8 }
+
+  def feed
+    following_ids_subselect = "SELECT followed_id FROM relationships
+                               WHERE  follower_id = :user_id"
+    Lesson.where("user_id IN (:following_ids_subselect)
```
Comment (chuanhtuan) [inline app/models/user.rb]: Lesson.where("user_id IN (#{following_ids_subselect})

**Pull request merged**